pkg/deploy/api/validation/validation.go

```diff
@@ -37,7 +37,7 @@ func ValidateDeploymentConfig(config *deployapi.DeploymentConfig) fielderrors.Va
 	if config.Spec.Replicas < 0 {
 		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec.replicas", config.Spec.Replicas, "replicas cannot be negative"))
 	}
-	if config.Spec.Selector == nil || len(config.Spec.Selector) == 0 {
+	if len(config.Spec.Selector) == 0 {
```

Comment:
I don't know if anything else can/will be done for further label customizations.
The changes here are exactly what happens in Kube right now (with the exception
of defaulting podtemplate labels to deployment.labels, not handled here since we
set defaults in the deployment.spec level and not deployment(.objectmeta))